_rules/visible-label-in-accessible-name-2ee8b8.md

```diff
@@ -118,61 +129,273 @@ This `button` element has the text "search" rendered as an magnifying glass icon
+
+The passes for two reasons: 1) because the ellipsis ("…") is [non-text content][], and 2) because the ellipsis is neither a letter nor a digit and so is filtered out by the [label in name algorithm][].
+
+```html
+<button aria-label="Next">Next…</button>
+```
+
+#### Passed Example 16
+
+This passes because the [label in name algorithm][] effectively ignores all punctuation and emoji, in both the visible inner text and the accessible name, as long as they don't break up words.
+
+```html
+<button aria-label="💡 Submit 💡">&gt;&gt;&gt; ** Submit ** &lt;&lt;&lt;</button>
+```
+
 ### Failed
 
 #### Failed Example 1
 
-This link has [visible][] text that is different from the [accessible name][].
+This link has [visible inner text][] that is very different from the [accessible name][].
 
 ```html
 <a href="https://act-rules.github.io/" aria-label="WCAG">ACT rules</a>
 ```
 
 #### Failed Example 2
 
-This button has [visible][] text that is only partially contained within the [accessible name][].
+This button has [visible inner text][] that is only partially contained within the [accessible name][].
 
 ```html
 <button aria-label="the full">The full label</button>
 ```
 
 #### Failed Example 3
 
-This link has [visible][] text with mathematical symbols, that does not match the [accessible name][] because the mathematical symbols were written out in the accessible name. This is [explicitly mentioned in WCAG](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name#mathematical-expressions-and-formulae).
+This button has [visible inner text][] that is fully contained within the [accessible name][] when viewed as a character-by-character substring. But that does not satisfy our [label in name algorithm][], which works on full words. So this fails the rule.
 
 ```html
-<a href="/" aria-label="Proof of two multiplied by two is four">Proof of 2&times;2=4</a>
+<a href="#" aria-label="Discover Italy">Discover It</a>
 ```
 
 #### Failed Example 4
 
-This link has [visible][] text does not match the [accessible name][] because there is a hyphen in the accessible name.
+This link's [accessible name][] contains two tokens (according to the[label in name algorithm][]) and the [visible inner text][] contains one token. So it fails the rule.
 
 ```html
-<a href="#" aria-label="non-standard">nonstandard</a>
+<a aria-label="just ice" href="#">justice</a>
 ```
 
 #### Failed Example 5
 
-This link has [visible][] text does not match the [accessible name][] because there are extra spaces in the accessible name.
+This link has an [accessible name][] which contains a hyphen. The [label in name algorithm][] breaks up words on hyphens. So it turns "non-standard" into two tokens: "non" and "standard". So this fails the rule.
+
+```html
+<a href="#" aria-label="non-standard">nonstandard</a>
+```
+
+#### Failed Example 6
+
+The rule has no special handling for acronyms or initialisms.
+
+```html
+<a aria-label="WCAG" href="#">W C A G</a>
+```
+
+#### Failed Example 7
```

Comment:
> The rule has no special handling for abbreviations.

This seems problematic. For many abbreviations it is much more natural (in English) to say the full word when encountering an abbreviations (for example, street abbreviations in an address like "Rd.", "St.", "Pl.", etc).

Perhaps it is just another shortcoming of this SC, but the calculation seems to me to need to be more fuzzy, and bias towards passing, not failing an abbreviation construct.